### PR TITLE
Medical record service

### PR DESCRIPTION
**Pull request by gmsafe**

### CODE REVIEW

File: opencdx-health/src/main/java/cdx/opencdx/health/repository/OpenCDXMedicalRecordRepository.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import io.micrometer.observation.annotation.Observed;
+import org.springframework.stereotype.Repository;
+
+/**
+ * Interface for the Database Medical Record model.
+ */
+@Repository
+@Observed(name = "opencdx")
+public interface OpenCDXMedicalRecordRepository extends OpenCDXRepository<OpenCDXMedicalRecordModel> {
+    //    /**
```
Comment (Ghost-Programmer) [inline opencdx-health/src/main/java/cdx/opencdx/health/repository/OpenCDXMedicalRecordRepository.java:29]: Remove the commented out code

File: opencdx-health/src/main/java/cdx/opencdx/health/service/impl/OpenCDXMedicalRecordProcessServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+ * Interface for the OpenCDXMedicalRecordProcessService
+ */
+@Slf4j
+@Service
+@Observed(name = "opencdx")
+public class OpenCDXMedicalRecordProcessServiceImpl implements OpenCDXMedicalRecordProcessService {
+
+    private final OpenCDXCurrentUser openCDXCurrentUser;
+    private final OpenCDXMedicalRecordRepository openCDXMedicalRecordRepository;
+    //private final OpenCDXMedicalRecordService openCDXMedicalRecordService;
```
Comment (Ghost-Programmer) [inline opencdx-health/src/main/java/cdx/opencdx/health/service/impl/OpenCDXMedicalRecordProcessServiceImpl.java:48]: Removed Commented out code

File: opencdx-health/src/main/java/cdx/opencdx/health/service/impl/OpenCDXMedicalRecordServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+            openCDXProfileModel = openCDXProfileRepository
+                    .findByNationalHealthId(request.getNationalHealthId())
+                    .orElseThrow(() -> new OpenCDXNotFound(this.getClass().getName(), 2, "NHID not found"));
+        }
+        OpenCDXMedicalRecordModel model =
+                this.openCDXMedicalRecordRepository.save(new OpenCDXMedicalRecordModel(openCDXProfileModel));
+
+        try {
+            OpenCDXIAMUserModel currentUser = this.openCDXCurrentUser.getCurrentUser();
+            this.openCDXAuditService.phiAccessed(
```
Comment (Ghost-Programmer) [inline opencdx-health/src/main/java/cdx/opencdx/health/service/impl/OpenCDXMedicalRecordServiceImpl.java:112]: Change to PHI created

File: opencdx-health/src/main/java/cdx/opencdx/health/service/impl/OpenCDXVaccineServiceImpl.java [hunk 4]
```diff
@@ -149,6 +150,39 @@ public Vaccine getVaccineById(GetVaccineByIdRequest request) {
...
+        OpenCDXVaccineModel vaccineModel = this.openCDXVaccineRepository
+                .findById(new OpenCDXIdentifier(request.getId()))
+                .orElseThrow(() -> new OpenCDXNotFound(
+                        DOMAIN,
+                        3,
+                        "FAILED_TO_FIND_VACCINE" + request.getId()));
+        OpenCDXVaccineModel model = this.openCDXVaccineRepository.save(vaccineModel.update(request));
+        try {
+            OpenCDXIAMUserModel currentUser = this.openCDXCurrentUser.getCurrentUser();
+            this.openCDXAuditService.phiCreated(
```
Comment (Ghost-Programmer) [inline opencdx-health/src/main/java/cdx/opencdx/health/service/impl/OpenCDXVaccineServiceImpl.java:165]: PHI Updated

**Review by Ghost-Programmer — CHANGES_REQUESTED** [review]
18 Sonarlint issues that need to be resolved as well.

File: opencdx-health/src/main/java/cdx/opencdx/health/repository/OpenCDXMedicalRecordRepository.java (hunk 1 above)
Comment (gmsafe) [reply opencdx-health/src/main/java/cdx/opencdx/health/repository/OpenCDXMedicalRecordRepository.java]: Done

File: opencdx-health/src/main/java/cdx/opencdx/health/service/impl/OpenCDXMedicalRecordProcessServiceImpl.java (hunk 2 above)
Comment (gmsafe) [reply opencdx-health/src/main/java/cdx/opencdx/health/service/impl/OpenCDXMedicalRecordProcessServiceImpl.java]: Done

File: opencdx-health/src/main/java/cdx/opencdx/health/service/impl/OpenCDXMedicalRecordServiceImpl.java (hunk 3 above)
Comment (gmsafe) [reply opencdx-health/src/main/java/cdx/opencdx/health/service/impl/OpenCDXMedicalRecordServiceImpl.java:110]: Done

File: opencdx-health/src/main/java/cdx/opencdx/health/service/impl/OpenCDXVaccineServiceImpl.java (hunk 4 above)
Comment (gmsafe) [reply opencdx-health/src/main/java/cdx/opencdx/health/service/impl/OpenCDXVaccineServiceImpl.java]: Done

**Pull request merged**